src/Relations/MorphToMany.php

```diff
@@ -0,0 +1,459 @@
+
+        return $instance;
+    }
+
+    /** @inheritdoc */
+    public function sync($ids, $detaching = true)
+    {
+        $changes = [
+            'attached' => [],
+            'detached' => [],
+            'updated' => [],
+        ];
+
+        if ($ids instanceof Collection) {
+            $ids = $this->parseIds($ids);
+        } elseif ($ids instanceof Model) {
+            $ids = $this->parseIds($ids);
+        }
+
+        // First we need to attach any of the associated models that are not currently
+        // in this joining table. We'll spin through the given IDs, checking to see
+        // if they exist in the array of current ones, and if not we will insert.
+        if ($this->getInverse()) {
+            $current = $this->extractIds($this->parent->{$this->table} ?: []);
+        } else {
+            $current = $this->parent->{$this->relatedPivotKey} ?: [];
+        }
+
+        // See issue #256.
+        if ($current instanceof Collection) {
+            $current = $this->parseIds($current);
+        }
+
+        $records = $this->formatSyncList($ids);
+
+        $current = Arr::wrap($current);
+
+        $detach = array_diff($current, array_keys($records));
+
+        // We need to make sure we pass a clean array, so that it is not interpreted
+        // as an associative array.
+        $detach = array_values($detach);
+
+        // Next, we will take the differences of the currents and given IDs and detach
+        // all of the entities that exist in the "current" array but are not in the
+        // the array of the IDs given to the method which will complete the sync.
+        if ($detaching && count($detach) > 0) {
+            $this->detach($detach);
+
+            $changes['detached'] = array_map(function ($v) {
+                return is_numeric($v) ? (int) $v : (string) $v;
+            }, $detach);
+        }
+
+        // Now we are finally ready to attach the new records. Note that we'll disable
+        // touching until after the entire operation is complete so we don't fire a
+        // ton of touch operations until we are totally done syncing the records.
+        $changes = array_merge(
+            $changes,
+            $this->attachNew($records, $current, false),
+        );
+
+        if (count($changes['attached']) || count($changes['updated'])) {
+            $this->touchIfTouching();
+        }
+
+        return $changes;
+    }
+
+    /** @inheritdoc */
+    public function updateExistingPivot($id, array $attributes, $touch = true)
+    {
+        // Do nothing, we have no pivot table.
+    }
+
+    /** @inheritdoc */
+    public function attach($id, array $attributes = [], $touch = true)
+    {
+        if ($id instanceof Model) {
+            $model = $id;
+
+            $id = $this->parseId($model);
+
+            if ($this->getInverse()) {
+                // Attach the new ids to the parent model.
+                $this->parent->push($this->table, [
```

Comment:
after any `sync` or `attach` action, I had to refresh the instance to fetch the correct relation's data and I couldn't find the problem. but finally, I figured out that when we push an array in unique mode, after the first time, only the id field was pushed into the instance pivot table (because id is unique). So, I solved this issue on the `push` and `pull` actions by wrapping the data in another array.